dbms/src/AggregateFunctions/AggregateFunctionHistogram.cpp

```diff
@@ -0,0 +1,36 @@
+#include "AggregateFunctionHistogram.h"
+#include "AggregateFunctionFactory.h"
+#include "FactoryHelpers.h"
+#include "Helpers.h"
+
+#include <Common/FieldVisitors.h>
+
+namespace DB {
+
+AggregateFunctionPtr createAggregateFunctionHistogram(const std::string & name, const DataTypes & arguments, const Array & params)
+{
+    if (params.size() != 1)
+    {
+        throw Exception("Function " + name + " requires only bins count");
```

Comment:
It's better to add ErrorCode.
It's not obvious that bins count is a parameter (not an argument), and we must specify it. For example:
```
:) select histogram(number) from (select * from system.numbers limit 20);

SELECT histogram(number)
FROM 
(
    SELECT *
    FROM system.numbers 
    LIMIT 20
) 

Received exception from server (version 1.1.54386):
Code: 0. DB::Exception: Received from localhost:9004, ::1. DB::Exception: Function histogram requires only bins count.
```